### PR TITLE
YA

### PR DESCRIPTION
**Pull request by Yuhans**

### CODE REVIEW

File: src/main/java/com/epam/javacourses2016/task3/SolverTask3.java [hunk 1]
```diff
@@ -17,6 +19,21 @@
...
+        for (Poem poem : poemsOfAuthor) {
+            List<String> poemsLines = poem.getLines();
+            allLines.addAll(poemsLines);
+        }
+        allLines.stream().sorted((string1, string2) -> {
+            if (string1.length() > string2.length()) {
+                return 1;
+            } else if (string1.length() < string2.length()) {
+                return -1;
+            } else return 0;
```
Comment (nkarulin) [inline src/main/java/com/epam/javacourses2016/task3/SolverTask3.java]: may we simplify this by
```java
return string2.length() - string1.length();
```

File: src/main/java/com/epam/javacourses2016/task9/SolverTask9.java [hunk 2]
```diff
@@ -17,6 +21,29 @@
      * @return Множество полученных слов.
      */
     public HashSet<String> getUniqueWords(File input) {
-        return null;
+        HashSet<String> result = new HashSet<>();
+        ArrayList<String> allWords = new ArrayList<>();
+        try {
+            FileReader fileInputStream = new FileReader(input);
```
Comment (nkarulin) [inline src/main/java/com/epam/javacourses2016/task9/SolverTask9.java]: Use try-with-resource, resources must be closed after use

File: src/main/java/com/epam/javacourses2016/task9/SolverTask9.java [hunk 3]
```diff
@@ -17,6 +21,29 @@
...
+            BufferedReader reader = new BufferedReader(fileInputStream);
+            String word;
+            while ((word = reader.readLine()) != null) {
+                allWords.add(word);
```
Comment (nkarulin) [inline src/main/java/com/epam/javacourses2016/task9/SolverTask9.java]: We don't need that collection, one single hash set with custom Comparator should be enough

**Review by nkarulin — COMMENTED** [review]
Nice Java 8 style! 
Please remove .gitignore and .class file from PR

**Comment by nkarulin** [comment]
Please enable tests

**Comment by Yuhans** [comment]
Don't understand how to use only one HashSet in Task_9. We can't add Comparator to constructor, it's able for TreeSet only.

**Comment by Yuhans** [comment]
Could you help me to find the answer, where can be the errors?) All checks have failed since i updated Idea.

**Comment by Yuhans** [comment]
Oh, the cause in tests, ok, i was too inattentive

File: src/main/java/com/epam/javacourses2016/task10/SolverTask10.java [hunk 4]
```diff
@@ -17,6 +20,21 @@
...
+                if(result.containsKey(word)) {
+                    result.put(word, result.get(word) + 1);
+                }
+                else {
+                    result.put(word, 1);
+                }
+            }
+        } catch (IOException e) {
+            System.out.println("I/O ex");
+            e.printStackTrace();
```
Comment (nkarulin) [inline src/main/java/com/epam/javacourses2016/task10/SolverTask10.java:36]: Better just rethrow it like `throw new RuntimeException(e)`

File: src/main/java/com/epam/javacourses2016/task4/SolverTask4.java [hunk 5]
```diff
@@ -28,7 +32,7 @@
      * @return Результат объединения множеств.
      */
     public Set<Integer> union(Set<Integer> first, Set<Integer> second) {
-        //TODO
-        return null;
+        Set<Integer> unionSet = Stream.concat(first.stream(), second.stream()).collect(Collectors.toSet());
```
Comment (nkarulin) [inline src/main/java/com/epam/javacourses2016/task4/SolverTask4.java:35]: :+1:  You'll like Scala or Clojure)

**Pull request merged**